api/task_delete.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// deleteTask deletes an existing task and its events. Does not delete
+// if the task is active.
+func (h *taskHandler) deleteTask(w http.ResponseWriter, r *http.Request) {
+	taskName, err := getTaskName(r.URL.Path, taskPath, h.version)
+	logger := logging.FromContext(r.Context()).Named(deleteTaskSubsystemName)
+	if err != nil {
+		logger.Trace("bad request", "error", err)
+		jsonErrorResponse(r.Context(), w, http.StatusBadRequest, err)
```

Comment:
I would look at the `api/handler.go sendError` method instead, this method conforms to the schema defined in the task lifecycle documentation we worked on and in the openapi.yaml.

I didn't change the behaviour of jsonErrorResponse because I really didn't want to touch any of the old api at this point.

Reply:
Also see one of the example handlers in task.go on how to extract the requestID from the context to pass to this method.

Reply:
Totally forgot about request id! 🤦‍♀️ I'm thinking now too we could return the request id as part of the delete response instead of the empty JSON it currently returns.